lib/kino/table.ex

```diff
@@ -108,36 +112,36 @@ defmodule Kino.Table do
   end
 
   def handle_event("order_by", %{"key" => nil}, ctx) do
-    {:noreply, ctx |> assign(order_by: nil, order: :asc) |> broadcast_update()}
+    {:noreply, ctx |> assign(order: nil) |> broadcast_update()}
   end
 
-  def handle_event("order_by", %{"key" => key_string, "order" => order}, ctx) do
-    order = String.to_existing_atom(order)
+  def handle_event("order_by", %{"key" => key_string, "direction" => direction}, ctx) do
+    direction = String.to_existing_atom(direction)
     key = lookup_key(ctx, key_string)
-    ctx = if key, do: assign(ctx, order_by: key, order: order), else: ctx
+    ctx = if key, do: assign(ctx, order: %{key: key, direction: direction}), else: ctx
     {:noreply, broadcast_update(ctx)}
   end
 
-  def handle_event("filter_by", %{"key" => key} = filter, ctx) do
-    filters = if ctx.assigns.filters, do: ctx.assigns.filters, else: []
-    idx = Enum.find_index(filters, &(&1["key"] == key))
-    updated_filters = if idx, do: List.replace_at(filters, idx, filter), else: filters ++ [filter]
-    keys = Enum.map(filters, &lookup_key(ctx, &1["key"])) |> Enum.all?()
-    ctx = if keys, do: assign(ctx, filters: updated_filters), else: ctx
+  def handle_event("filter_by", %{"key" => key_string, "filter" => filter, "value" => value}, ctx) do
+    updated_filters =
+      Enum.reject(ctx.assigns.content.filters, &(&1.key == key_string))
+      |> Kernel.++([%{key: key_string, filter: filter, value: value}])
+      |> Enum.map(&%{&1 | key: lookup_key(ctx, &1.key)})
+
+    ctx = if lookup_key(ctx, key_string), do: assign(ctx, filters: updated_filters), else: ctx
```

Comment:
Let's move `updated_filters =` inside the if, then we only need one `lookup_key` in total.